language/move-prover/bytecode/src/access_path.rs

```diff
@@ -109,6 +109,12 @@ pub trait AccessPathMap<T: AbstractDomain> {
     fn get_access_path(&self, ap: AccessPath) -> Option<&T>;
 }
 
+/// Trait for a domain that can be viewed as a partial map from access paths to values
+/// and values can be deleted using their access paths
+pub trait AccessPathDeletableMap<T: AbstractDomain> {
```

Comment:
Wondering about the thought process for a separate trait vs adding this as a function in `AccessPathMap`.

Reply:
The only reason I separated this to another trait was in case we might want an immutable `AccessPathMap` but I'm happy to move the function to `AccessPathMap` if you think we are not going to need this

Reply:
Ah, I see. I think moving it to `AccessPathMap` and using (e.g.) `&AccessPathMap` in type signatures where immutability is desired would probably make sense.